libs/backend/src/scan/cast_vote_records/export.ts

```diff
@@ -118,181 +205,201 @@ async function* getCastVoteRecordGenerator({
         pages: [
           {
             interpretation: front,
-            inlineBallotImage:
-              frontHasWriteIns && imageOptions.includeInlineBallotImages
-                ? await getInlineBallotImage(frontFilename)
-                : undefined,
             imageFileUri:
-              imageOptions.includedImageFileUris === 'all' ||
-              (imageOptions.includedImageFileUris === 'write-ins' &&
-                frontHasWriteIns)
-                ? `file:./${imageOptions.imagesDirectory}/${basename(
-                    frontFilename
-                  )}`
+              imageOptions.which === 'all' ||
+              (imageOptions.which === 'write-ins' && frontHasWriteIns)
+                ? `file:./${imageOptions.directory}/${basename(frontFilename)}`
                 : undefined,
           },
           {
             interpretation: back,
-            inlineBallotImage:
-              backHasWriteIns && imageOptions.includeInlineBallotImages
-                ? await getInlineBallotImage(backFilename)
-                : undefined,
             imageFileUri:
-              imageOptions.includedImageFileUris === 'all' ||
-              (imageOptions.includedImageFileUris === 'write-ins' &&
-                backHasWriteIns)
-                ? `file:./${imageOptions.imagesDirectory}/${basename(
-                    backFilename
-                  )}`
+              imageOptions.which === 'all' ||
+              (imageOptions.which === 'write-ins' && backHasWriteIns)
+                ? `file:./${imageOptions.directory}/${basename(backFilename)}`
                 : undefined,
           },
         ],
         ballotPageLayoutsLookup,
-      });
-    }
+      })
+    );
   }
-}
 
-async function* concatStreams(readables: NodeJS.ReadableStream[]) {
-  for (const readable of readables) {
-    for await (const chunk of readable) {
-      yield chunk;
-    }
-  }
-}
-
-/**
- * JSON format requires that we have no trailing commas, so we must
- * post-process the cast vote records iterator to add a comma after all but
- * the last.
- */
-async function* commaSeparateStringGenerator(
-  asyncGenerator: AsyncGenerator<string>
-): AsyncGenerator<string> {
-  let previous: Optional<string>;
-  for await (const current of asyncGenerator) {
-    if (previous) {
-      yield `${previous},`;
-    }
-    previous = current;
-  }
+  const castVoteRecordReportWithoutCastVoteRecords =
+    buildCastVoteRecordReportWithoutCastVoteRecords({
+      election,
+      electionId,
+      generatingDeviceId: scannerId,
+      scannerIds: [scannerId],
+      reportTypes: [CVR.ReportType.OriginatingDeviceExport],
+      isTestMode,
+      batchInfo,
+    });
 
-  // yield last element without trailing comma
-  if (previous) {
-    yield previous;
-  }
+  return ok(
+    streamifyCastVoteRecordReport({
```

Comment:
Is there a reason we're still building the report metadata and the cast vote records and combining them in a stream if we're just doing it all in memory? Seems like it might be simpler to just build the full report object and then stringify.

Reply:
Our max string size is like 0.5GB. With inline ballot images, we're limited to like 1,000 ballots if all ballots have images (scaled down). Without inline ballot images, that goes up to way higher, like 100,000. So I thought about just not worrying about it. But I think it's conceivable that we change our CVRs to be even more information dense in the future and I didn't want to bank on it all fitting in the one string.

Reply:
Ah ok, so there's a max string size that's diff from the process memory limit? That's helpful context. I think it's worth documenting some of this thinking in explanatory comments, since it's kind of an unconventional halfway between full streamed CVRs and in-memory CVRs

Or just going full streaming would be another option.

Reply:
Exactly, max string size is much more of a limit than what we can potentially have in an array or an object. Full streaming was hard because I was doing the sheet callback, but now that I'm making two passes, I'll see if it's easy to do full streaming.